pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -15,39 +15,73 @@
 package bootstrap
 
 import (
+	"bytes"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pkg/util"
+	"istio.io/pkg/env"
 
 	"istio.io/istio/pkg/kube/inject"
 	"istio.io/pkg/log"
 )
 
+var (
+	webhookConfigName = env.RegisterStringVar("WEBHOOK", "",
```

Comment:
By hardcoding a name - but istiod supports multi-tenant, i.e. you can have as many istiod-canary or istiod-foobar that you want, and each will work.

Reply:
I can derive it from the ISTIOD_ADDR - again, this is the first version and I'm using env variables because it's not yet clear what is the proper interface. Once we agree on an API - and get some 
feedback ( which is why I'm trying to get istiod working ASAP ) - we can remove the env.